tests/Unit/NumericalAlgorithms/Spectral/Test_ChebyshevGauss.cpp

```diff
@@ -0,0 +1,90 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "tests/Unit/TestingFramework.hpp"
+
+#include <cstddef>
+#include <utility>
+
+#include "DataStructures/DataVector.hpp"
+#include "NumericalAlgorithms/Spectral/Spectral.hpp"
+
+namespace Spectral {
+// To test the \f$w_k\f$ as opposed to the Spectral::quadrature_weights. We
+// don't provide a public accessor for the former since we only ever need them
+// in Spectral.cpp.
+template <Basis, Quadrature>
+std::pair<DataVector, DataVector> compute_collocation_points_and_weights(
+    size_t) noexcept;
+}  // namespace Spectral
+
+namespace {
+
+void test_points_and_weights(const size_t num_points,
+                             const DataVector expected_points,
+                             const DataVector expected_weights) {
+  const auto& points =
+      Spectral::collocation_points<Spectral::Basis::Chebyshev,
+                                   Spectral::Quadrature::Gauss>(num_points);
+  CHECK_ITERABLE_APPROX(expected_points, points);
+  // We test the \f$w_k\f$ here directly. Test_Spectral.cpp and
+  // Test_DefiniteIntegral.cpp take care of testing the
+  // Spectral::quadrature_weights.
+  const auto weights =
+      Spectral::compute_collocation_points_and_weights<
+          Spectral::Basis::Chebyshev, Spectral::Quadrature::Gauss>(num_points)
+          .second;
+  CHECK_ITERABLE_APPROX(expected_weights, weights);
+}
+
+}  // namespace
+
+SPECTRE_TEST_CASE("Unit.Numerical.Spectral.ChebyshevGauss.PointsAndWeights",
+                  "[NumericalAlgorithms][Spectral][Unit]") {
+  // Compare Chebyshev-Gauss points and weights to the analytic expressions
+  // \f$\x_j=-\cos{\frac{2j+1}{2p+2}\pi}\f$ and \f$w_k=\frac{\pi}{N}\f$, where
+  // \f$p=N-1\f$ is the polynomial degree, \f$N\f$ the number of collocation
+  // points and \f$0\leq j \leq p\f$ (\cite Kopriva, eqn. (1.128)).
```

Comment:
It would be best to cite a proper bibtex entry, so putting the reference in the namespace documentation sound like a good idea! Then it‘s actually parses by Doxygen so it shows up as a citation: https://www.stack.nl/~dimitri/doxygen/manual/commands.html#cmdcite

I‘d like to refer to the particular algorithms in the cpp files too, so how about there I leave the `\cite`s and update them with the correct bibtex entry?

Reply:
I think adding it to the namespace is good enough. In the cpp file just refer to it as `Kopriva algorithm blah`. If you really want something more direct in the cpp file do the DOI think. Nothing more is needed.